scalafix-core/shared/src/main/scala/scalafix/internal/rule/SingleAbstractMethod.scala

```diff
@@ -0,0 +1,273 @@
+          if (iterator.hasNext) {
+            slot = Some(iterator.next())
+          } else {
+            exhausted = true
+            slot = None
+          }
+        }
+      }
+      override def hasNext: Boolean = {
+        if (exhausted) {
+          false
+        } else {
+          if (slot.isDefined) true
+          else iterator.hasNext
+        }
+      }
+      def peek(): Option[T] = {
+        fill()
+        if (exhausted) None
+        else slot
+      }
+      def next: T = {
+        if (!hasNext) throw new NoSuchElementException()
+        val out = slot.getOrElse(iterator.next)
+        slot = None
+        out
+      }
+    }
+
+    class PatchBuilder(tokens: Tokens) {
+      private val iterator = new PeekableIterator(tokens.iterator)
+      private val patches = List.newBuilder[Patch]
+      def next: Token = iterator.next
+      def find[T <: Token: TokenInfo]: Option[Token] =
+        iterator.find(_.is[T](implicitly[TokenInfo[T]]))
+      def find(f: Token => Boolean): Option[Token] =
+        iterator.find(f)
+      def remove[T <: Token: TokenInfo]: Unit =
+        doRemove(find[T])
+      def remove(t: Token): Unit =
+        patches += ctx.removeToken(t)
+      def removeOptional[T <: Token: TokenInfo]: Unit =
+        if (iterator.peek.forall(_.is[T])) remove[T]
+      def removeLast[T <: Token: TokenInfo]: Unit = {
+        var last: Option[Token] = None
+        while (iterator.hasNext) {
+          last = find[T]
+        }
+        doRemove(last)
+      }
+      def addRight[T <: Token: TokenInfo](toAdd: String): Unit =
+        doOp(find[T], tt => patches += ctx.addRight(tt, toAdd))
+      def addLeft(t: Token, what: String): Unit =
+        patches += ctx.addLeft(t, what)
+      def doRemove[T <: Token](t: Option[Token])(
+          implicit ev: TokenInfo[T]): Unit =
+        doOp(t, tt => patches += ctx.removeToken(tt))
+      def doOp[T <: Token](t: Option[Token], op: Token => Unit)(
+          implicit ev: TokenInfo[T]): Unit =
+        t match {
+          case Some(t) => op(t)
+          case _ =>
+            throw new Exception(
+              s"""|cannot find ${ev.name}
+                  |Tokens:
+                  |$tokens""".stripMargin
+            )
+        }
+      def result(): Patch = Patch.fromIterable(patches.result())
+    }
+
+    def patchSam(
+        builder: PatchBuilder,
+        mods: List[Mod],
+        keepClassName: Boolean): Patch = {
+      import builder._
+
+      mods.foreach(_.tokens.foreach(remove))
+
+      // new X(){def m(a: A, b: B, c: C): D = <body> }
+      remove[KwNew]
+      remove[Space]
+      if (!keepClassName) {
+        remove[Ident]
+      } else {
+        addRight[Ident](" =") // class name
+      }
+      removeOptional[LeftParen]
+      removeOptional[RightParen]
+
+      remove[LeftBrace]
+      remove[KwDef]
+      remove[Space]
+      remove[Ident] // method name
+
+      def patchParams(hasParams: Boolean): (Token, Boolean) = {
+        find(
```

Comment:
Please use `find { t => ... }`. I should make scalafmt replace `(` with `{` for large arguments!!